MathLib/LinAlg/MatrixVectorTraits.cpp

```diff
@@ -84,7 +92,32 @@ std::unique_ptr<PETScMatrix>
 MatrixVectorTraits<PETScMatrix>::
 newInstance(MatrixSpecifications const& spec)
 {
-    return std::unique_ptr<PETScMatrix>(new PETScMatrix(spec.nrows)); // TODO sparsity pattern
+    auto const nrows = spec.dof_table ? spec.dof_table->dofSizeLocal() : spec.nrows;
+    auto const ncols = spec.dof_table ? nrows : spec.ncols;
+
+    // TODO I guess it is not hard to make AssemblerLib::computeSparsityPattern()
```

Comment:
I think, we should go for precise preallocation in the future, since in our FEM processes we can easily access that information anyway.